HISTORY.md

```diff
@@ -2,6 +2,8 @@
 ###### ????-??-??
   * Add adaptive max pooling and adaptive mean pooling layers (#2195).
 
+  * Add copy constructor in all layers of ANN (#2325).
```

Comment:
Ahhh, so this is really tedious, but I realized that if we are adding a copy constructor, we also need to add a copy operator, and then also a move constructor and move operator (maybe for those last two, sometimes the `default` implementation can be specified).  If you prefer, I think it's okay to open a separate issue for each of those, since it would make this PR even bigger than it already is (and it's kind of time-consuming to do all that!).  However we should definitely ensure that we handle it.

Reply:
Yeah they need to be handled but we add them into another issue because it will unnecessary to add more code in this pr.